cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsTerminateService.java

```diff
@@ -112,12 +112,9 @@ private void waitAndDeleteCloudformationStack(AuthenticatedContext ac, CloudStac
         resumeAutoScalingPolicies(ac, stack);
         LOGGER.debug("Delete cloudformation stack from resources");
         DeleteStackRequest deleteStackRequest = new DeleteStackRequest().withStackName(cFStackName);
-        cfRetryClient.deleteStack(deleteStackRequest);
-        Waiter<DescribeStacksRequest> stackDeleteCompleteWaiter = amazonCloudFormationClient.waiters().stackDeleteComplete();
         try {
-            WaiterParameters<DescribeStacksRequest> describeStacksRequestWaiterParameters = new WaiterParameters<>(describeStacksRequest)
-                    .withPollingStrategy(getBackoffCancellablePollingStrategy(null));
-            stackDeleteCompleteWaiter.run(describeStacksRequestWaiterParameters);
+            retryService.testWith2SecDelayMax5Times(() -> isStackDeleted(cfRetryClient, amazonCloudFormationClient,
```

Comment:
It was intentional, yes. There are two things being retried. The `cfRetryClient.deleteStack` method is retrying the initiation of the request; if the initial request is rejected or fails immediately, it will retry up to 15 time to initiate it again. However, once it's successfully initiated, that retry is complete and there's nothing left to do. The purpose of the `isStackDeleted` method is to check the final state of the deletion, and allow us to retry if the final state is a failure. So one retry checks initialization state, and one checks the final state.

Reply:
Talked this out with Tamas and added additional logging messages to make the logic of this section more cler.